test/shared/PosmTestSetup.sol

```diff
@@ -24,12 +25,26 @@ contract PosmTestSetup is Test, Deployers, DeployPermit2, LiquidityOperations {
     HookSavesDelta hook;
     address hookAddr = address(uint160(Hooks.AFTER_ADD_LIQUIDITY_FLAG | Hooks.AFTER_REMOVE_LIQUIDITY_FLAG));
 
+    HookModifyLiquidities hookModifyLiquidities;
+    address hookModifyLiquiditiesAddr =
+        address(uint160(Hooks.BEFORE_SWAP_FLAG | Hooks.AFTER_ADD_LIQUIDITY_FLAG | Hooks.AFTER_REMOVE_LIQUIDITY_FLAG));
```

Comment:
it only has a before swap hook?

Reply:
it inherits HookSavesDelta

i wrote a test where if a hook burns a position, it _should_ have the principal capital of the original position. so needed the hook-delta-saving functionality